modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPSpecificationOptionLocalServiceImpl.java

```diff
@@ -235,16 +235,21 @@ public CPSpecificationOption updateCPOptionCategoryId(
 	@Indexable(type = IndexableType.REINDEX)
 	@Override
 	public CPSpecificationOption updateCPSpecificationOption(
-			long cpSpecificationOptionId, long cpOptionCategoryId,
-			long listTypeDefinitionId, Map<Locale, String> titleMap,
-			Map<Locale, String> descriptionMap, boolean facetable, String key,
-			double priority, ServiceContext serviceContext)
+			String externalReferenceCode, long cpSpecificationOptionId,
```

Comment:
I don't know if we want to modify this method or add the addOrUpdate. I can find this solution also in other places (like [ListTypeDefinitionLocalServiceImpl](https://github.com/liferay/liferay-portal/blob/c3c6410fdd9089318054c83c4b2aa7cae0228077/modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/impl/ListTypeDefinitionLocalServiceImpl.java#L142) or [ListTypeEntryLocalServiceImpl](https://github.com/liferay/liferay-portal/blob/97db3371b6846eb9bd39f3102dcf267586511235/modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/impl/ListTypeEntryLocalServiceImpl.java#L188) or [CommerceOrderTypeLocalServiceImpl](https://github.com/liferay/liferay-portal/blob/6ce0b9eb049e99a50285a324a3bd4afd76d340ab/modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderTypeLocalServiceImpl.java#L240)). @andrea-ale-sbarra what do you think?